app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,110 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
+  before_action :load_and_authorize_posts, except: [:create, :reply]
+
+  def create
+    if generate_post
+      redirect_to create_success_path
+    else
+      redirect_to create_fail_path
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @post.update_attributes(posts_params)
+      redirect_to update_success_path
+    else
+      redirect_to update_fail_path
+    end
+  end
+
+  def destroy
+    if @post.destroy
+      redirect_to destroy_success_path
+    else
+      redirect_to destroy_fail_path
+    end
+  end
+
+  def reply
```

Comment:
This is for replying to a post without js. It will redirect to a new page called `reply`.